lib/matplotlib/backends/backend_gtk3agg.py

```diff
@@ -48,14 +50,28 @@ def on_draw_event(self, widget, ctx):
                 incref=True)
 
         for bbox in bbox_queue:
-            area = self.copy_from_bbox(bbox)
-            buf = np.fromstring(area.to_string_argb(), dtype='uint8')
-
             x = int(bbox.x0)
             y = h - int(bbox.y1)
             width = int(bbox.x1) - int(bbox.x0)
             height = int(bbox.y1) - int(bbox.y0)
 
+            buf = (np.fromstring(self.copy_from_bbox(bbox).to_string_argb(),
+                                 dtype='uint8')
+                   .reshape((width, height, 4)))
+            # cairo wants premultiplied alpha.  Only bother doing the
+            # conversion when the alpha channel is not fully opaque, as the
+            # cost is not negligible.
+            if sys.byteorder == "little":
+                alpha = buf[..., -1:]
+                if alpha.min() != 0xff:
+                    buf = (np.dstack([buf[..., :3] * (alpha / 0xff), alpha])
```

Comment:
Do you need to make a copy? Isn't `buf` an independent thing, so can you do an in-place multiply?

Reply:
was a bit more tricky to write correctly but fixed it